Yafc/Widgets/ImmediateWidgets.cs

```diff
@@ -205,11 +205,11 @@ public static class ImmediateWidgets {
         /// <param name="amount">Display this value, formatted appropriately for <paramref name="unit"/>.</param>
         /// <param name="unit">Use this unit of measure when formatting <paramref name="amount"/> for display.</param>
         /// <param name="useScale">If <see langword="true"/>, this icon will be displayed at <see cref="ProjectPreferences.iconScale"/>, instead of at 100% scale.</param>
-        public static Click BuildFactorioObjectWithAmount(this ImGui gui, FactorioObject? goods, float amount, UnitOfMeasure unit, SchemeColor bgColor = SchemeColor.None, SchemeColor textColor = SchemeColor.None, bool useScale = true) {
+        public static Click BuildFactorioObjectWithAmount(this ImGui gui, FactorioObject? goods, float amount, UnitOfMeasure unit, SchemeColor bgColor = SchemeColor.None, SchemeColor textColor = SchemeColor.None, bool useScale = true, ObjectTooltipOptions tooltipOptions = default) {
```

Comment:
I have a feeling that the amount of things we pass as parameters hurts the readability. I'm thinking about fixing it with something like a [Context](https://www.baeldung.com/cs/context-design-pattern) pattern or a DI framework. I briefly considered putting all the parameters that we ever pass in a single object, but it is bad practice, so I'll need to read on how to do it properly.

For this PR, I don't ask for a refactor. I'm just thinking aloud that the problem is present.